kyverno/kyverno
# feat: oci pull/push support for policie(s)

Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

## Explanation



To test this:

1. Create a local registry with one of the following services: 
   * _docker run -d -p 5000:5000 --restart=always --name registry registry:2_
   		* https://docs.docker.com/registry/deploying/#run-a-local-registry
   * _docker run -d -p 5000:5000 ghcr.io/project-zot/zot-linux-amd64:latest_
   		* https://zotregistry.io 
2. Copy of the policy from Kyverno policies repository: _curl -sSLO https://github.com/kyverno/policies/raw/main/other/require-base-image/require-base-image.yaml_
3. Build the source code from that branch
4. Use `oci push` command to push policy: _./kyverno oci -i localhost:5001/hello-world-policy:latest push --policy disallow-capabilities.yaml_ (⚠️you can point to a directory instead of pointing directly to a policy file)
5. If everything works well, remove the policy file and pull it from an OCI registry: _./kyverno oci -i localhost:5001/hello-world-policy:latest pull --directory ._
6. Check whether the given policy exists locally

/cc @JimBugwadia @chipzoller @eddycharly @realshuting 

## Related issue



Implements https://github.com/kyverno/KDP/pull/19

## Milestone of this PR


## What type of PR is this



/kind feature

## Proposed Changes



### Proof Manifests



## Checklist



  
  
  
  

## Further Comments
